plugins/volumeFilter.go

```diff
@@ -174,47 +179,55 @@ func volumeFilterFn(dailyOTB *VolumeFilterConfig, dailyTBBAccumulator *VolumeFil
 	}
 
 	if isSell {
-		opToReturn := op
-		newAmountBeingSold := amountValueUnitsBeingSold
-		var keepSellingBase bool
-		var keepSellingQuote bool
+		// initialize the string for new amounts and logging
+		newAmount := ""
+		newAmountLog := ""
+		var newAmountBeingSold float64
 		if lp.baseAssetCapInBaseUnits != nil {
-			projectedSoldInBaseUnits := *dailyOTB.BaseAssetCapInBaseUnits + *dailyTBBAccumulator.BaseAssetCapInBaseUnits + amountValueUnitsBeingSold
-			keepSellingBase = projectedSoldInBaseUnits <= *lp.baseAssetCapInBaseUnits
-			newAmountString := ""
-			if lp.mode == volumeFilterModeExact && !keepSellingBase {
-				newAmount := *lp.baseAssetCapInBaseUnits - *dailyOTB.BaseAssetCapInBaseUnits - *dailyTBBAccumulator.BaseAssetCapInBaseUnits
-				if newAmount > 0 {
-					newAmountBeingSold = newAmount
-					opToReturn.Amount = fmt.Sprintf("%.7f", newAmountBeingSold)
-					keepSellingBase = true
-					newAmountString = ", newAmountString = " + opToReturn.Amount
-				}
+			newAmountBeingSold = getRemainingCapacity(
+				*dailyOTB.BaseAssetCapInBaseUnits,
+				*dailyTBBAccumulator.BaseAssetCapInBaseUnits,
+				*lp.baseAssetCapInBaseUnits,
+				amountValueUnitsBeingSold,
+				1.0, // we don't want to apply the sell price to the base amount
+			)
+
+			// if the amount changed, update the amount and the log
+			if newAmountBeingSold != amountValueUnitsBeingSold {
+				newAmount = fmt.Sprintf("%.7f", newAmountBeingSold)
+				newAmountLog = fmt.Sprintf(", newAmountString = %s", newAmount)
 			}
-			log.Printf("volumeFilter:  selling (base units), price=%.8f amount=%.8f, keep = (projectedSoldInBaseUnits) %.7f <= %.7f (config.BaseAssetCapInBaseUnits): keepSellingBase = %v%s", sellPrice, amountValueUnitsBeingSold, projectedSoldInBaseUnits, *lp.baseAssetCapInBaseUnits, keepSellingBase, newAmountString)
-		} else {
-			keepSellingBase = true
+
+			// TODO DS Confirm that this is the desired approach to logging.
```

Comment:
The approach to logging is my only real open question. I tried to include all the information in the existing logs, excluding `keepSellingBase`/`keepSellingQuote`, as those variables no longer exist. This required computing `projectedSoldInBaseUnits` here as well as in the new extracted function. I considered logging in the extracted function, but that pretty clearly solves it at the wrong level. Curious if you have thoughts here - should I change logging entirely?